From a51e1a7583328d3bbd65293a36c36c3091c34dfb Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Wed, 8 Jul 2015 10:10:50 -0700
Subject: [PATCH 2/3] Find SYSROOT relative to its own location

Rebase "default-sysroot.patch" to make the update-ca-certificates
script relocatable.

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 sbin/update-ca-certificates | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/sbin/update-ca-certificates b/sbin/update-ca-certificates
index 3402039..b3452cb 100755
--- a/sbin/update-ca-certificates
+++ b/sbin/update-ca-certificates
@@ -24,6 +24,40 @@
 verbose=0
 fresh=0
 default=0
+
+if [ -z "$SYSROOT" ]; then
+  local_which () {
+    if [ $# -lt 1 ]; then
+      return 1
+    fi
+
+    (
+      IFS=:
+      for entry in $PATH; do
+        if [ -x "$entry/$1" ]; then
+          echo "$entry/$1"
+          exit 0
+        fi
+      done
+      exit 1
+    )
+  }
+
+  case "$0" in
+    */*)
+      sbindir=$(cd ${0%/*} && pwd)
+      ;;
+    *)
+      sbindir=$(cd $(dirname $(local_which $0)) && pwd)
+      ;;
+  esac
+  prefix=${sbindir%/*}
+  SYSROOT=${prefix%/*}
+  if [ ! -d "$SYSROOT/usr/share/ca-certificates" ]; then
+    SYSROOT=
+  fi
+fi
+
 CERTSCONF=$SYSROOT/etc/ca-certificates.conf
 CERTSDIR=$SYSROOT/usr/share/ca-certificates
 LOCALCERTSDIR=$SYSROOT/usr/local/share/ca-certificates
-- 
2.1.4

